pkg/controllers/reconciler_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controllers
+
+import (
+	"testing"
+)
+
+func TestUpgrade(t *testing.T) {
+	const (
+		currentImage     = "intel/intel-dsa-plugin:0.23.0"
+		currentInitimage = "intel/intel-idxd-config-initcontainer:0.23.0"
+	)
+
+	image := "intel/intel-dsa-plugin:0.22.0"
+	initimage := "intel/intel-idxd-config-initcontainer:0.22.0"
+
+	expectedImage := currentImage
+	expectedInitimage := currentInitimage
+
+	upgrade := UpgradeImages(&image, &initimage)
+
+	if !(upgrade == true && image == expectedImage && initimage == expectedInitimage) {
+		t.Errorf("expectedImage: %s, received: %s", expectedImage, image)
+		t.Errorf("expectedInitimage: %s, received: %s", expectedInitimage, initimage)
+	}
+
+	image = currentImage
+	initimage = currentInitimage
+
+	expectedImage = currentImage
+	expectedInitimage = currentInitimage
+
+	upgrade = UpgradeImages(&image, &initimage)
+
+	if !(upgrade == false && image == expectedImage && initimage == expectedInitimage) {
+		t.Errorf("expectedImage: %s, received: %s", expectedImage, image)
+		t.Errorf("expectedInitimage: %s, received: %s", expectedInitimage, initimage)
+	}
+
+	image = "intel/intel-dsa-plugin:latest"
+	initimage = "intel/intel-idxd-config-initcontainer:latest"
+
+	expectedImage = "intel/intel-dsa-plugin:latest"
+	expectedInitimage = "intel/intel-idxd-config-initcontainer:latest"
+
+	upgrade = UpgradeImages(&image, &initimage)
+
+	if !(upgrade == false && image == expectedImage && initimage == expectedInitimage) {
+		t.Errorf("expectedImage: %s, received: %s", expectedImage, image)
+		t.Errorf("expectedInitimage: %s, received: %s", expectedInitimage, initimage)
+	}
+
+	image = "intel/intel-dsa-plugin"
+	initimage = "intel/intel-idxd-config-initcontainer"
+
+	expectedImage = "intel/intel-dsa-plugin"
+	expectedInitimage = "intel/intel-idxd-config-initcontainer"
+
+	upgrade = UpgradeImages(&image, &initimage)
+
+	if !(upgrade == false && image == expectedImage && initimage == expectedInitimage) {
+		t.Errorf("expectedImage: %s, received: %s", expectedImage, image)
+		t.Errorf("expectedInitimage: %s, received: %s", expectedInitimage, initimage)
+	}
+}
```

Comment:
Nit: this looks like a good fit for a table-driven test (e.g. like in cmd/fpga_plugin/fpga_plugin_test.go).